File: javascript/task_manager.js
const Task = require('./task');
const { loadTasks, saveTasks } = require('./storage');
const { is_valid_task_name, is_valid_due_date } = require('./validator');

const TASK_FILE = 'tasks.json';

function addTask(name, due_date, task_file = TASK_FILE) {
  const tasks = loadTasks(task_file);
  is_valid_task_name(name, tasks);
  is_valid_due_date(due_date);
  tasks.push(new Task(name, due_date));
  saveTasks(tasks, task_file);
}

function markTaskAsDone(name, task_file = TASK_FILE) {
  const tasks = loadTasks(task_file);
  let found = false;
  for (const task of tasks) {
    if (task.name === name) {
      task.status = 'Done';
      found = true;
      break; // Exit the loop since the task is found
    }
  }

  if (!found) {
    console.log(`${name} not found`); //F strings are ${} in JavaScript
  } else {
    saveTasks(tasks, task_file);
  }
}

function getTasks(task_file = TASK_FILE) {
  const tasks = loadTasks(task_file);
  for (const task of tasks) {
    console.log(task);
  }
}

module.exports = { addTask, markTaskAsDone, getTasks};

File: python/task_manager.py
from task import Task
from storage import load_tasks, save_tasks
from validator import is_valid_task_name, is_valid_due_date

TASK_FILE = "tasks.json"

def add_task(name, due_date, task_file=TASK_FILE):
    """
    Add a task to the list of tasks
    """
    tasks = load_tasks(task_file)
    is_valid_task_name(name, tasks)
    is_valid_due_date(due_date)
    tasks.append(Task(name, due_date))
    save_tasks(tasks, task_file)

def mark_task_as_done(name, task_file=TASK_FILE):
    """
    Mark a task as done
    """
    tasks = load_tasks(task_file)
    found = False
    for task in tasks:
        if task.name == name:
            task.status = "Done"
            found = True
    
    if not found:
        print(f"{name} not found")
    else:
        save_tasks(tasks, task_file)

def get_tasks(task_file=TASK_FILE):
    """
    Get all tasks
    """
    tasks = load_tasks(task_file)
    for task in tasks:
        print(task)
